specification/protocol/otlp.md

```diff
@@ -145,16 +148,47 @@ was not delivered.
 
 #### OTLP/gRPC Response
 
-The server may respond with either a success or an error to the requests.
+The response MUST be the appropriate serialized Protobuf message (see below for
+the specific message to use in the [Success](#success),
+[Partial Success](#partial-success) and [Failure](#failures) cases).
+
+##### Success
 
 The success response indicates telemetry data is successfully processed by the
 server. If the server receives an empty request (a request that does not carry
 any telemetry data) the server SHOULD respond with success.
 
-Success response is returned via
-[Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto)
-message (`ExportTraceServiceResponse` for traces, `ExportMetricsServiceResponse`
-for metrics, `ExportLogsServiceResponse` for logs).
+On success, the server response MUST be
```

Comment:
I wonder if we can enforce a MUST here. I'd try, but I could see this becoming a sticky point here.

Reply:
It was already a MUST for HTTP, I think it should be the same enforcement. We'll see what the others say about it I guess.

Reply:
I just wonder if we can force them to use the details. I can imagine people just wanting to send 200, but not put the number of lines (for performance reasons, for example). That's why I think this `MUST` is maybe not ideal.

Reply:
Ah I thought you were talking about the "total success" case, and not the partial success, since the comment is on that line. 

Fair enough, I think I can change MUST to SHOULD for the partial success case. It's not mandatory and receivers that are not interested can just avoid it. I think it also addresses what @dyladan  said below about receivers that are under stress or the ones that do actual processing asynchronously.

Reply:
Ok I think we are talking past each other. When a server ingests all data without any issue, it needs to return the empty `Export*ServiceResponse` to clients/senders. It's basically an empty body response there's no cost or any performance penalty for servers/clients in this case. As well, it was already a MUST for HTTP, I think this was just a oversight in the gRPC part.

For the partial success it's different and people _could_ put whatever there and it could grow, or as @dyladan said, not all receivers may want/can do this so I will change to SHOULD.

Reply:
To circle back, I did some changes in https://github.com/dynatrace-oss-contrib/opentelemetry-specification/commit/2f29c83a9e86d9c7b1f21ae0e75a73d5422da0b8 so now hopefully it's all cleared up.  The reply to this other thread is somewhat linked/relevant https://github.com/dynatrace-oss-contrib/opentelemetry-specification/pull/20#discussion_r907308680